packages/cli/src/transformers.ts

```diff
@@ -547,6 +549,110 @@ export function getTypeImportExportTransformer(_context: TransformerOptions) {
   };
 }
 
+/**
+ * The options for the {@link getImportAttributeTransformer} function.
+ */
+export type ImportAssertionTransformerOptions = {
+  /**
+   * The type of the module, i.e., CommonJS or ES module, to apply the
+   * transformation to.
+   */
+  moduleType: FileFormat;
+
+  /**
+   * The import assertion type to apply.
+   */
+  type: string;
+};
+
+/**
+ * Get a transformer that adds an import attribute to the given module type with
+ * the given type attribute. This is mainly useful for JSON imports, which
+ * require `with { type: 'json' }` to be added to the import statement.
```

Comment:
Is the `with` syntax supported universally? Does it work in all the bundlers we use etc?

Reply:
All the popular bundlers, TypeScript, V8/Node/Chrome, Firefox, etc. seem to support it now. It's only used for the ESM build in the first place, which assumes a modern bundler.